components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementListener.java

```diff
@@ -0,0 +1,621 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.role.v2.mgt.core.listener;
+
+import org.wso2.carbon.identity.role.v2.mgt.core.GroupBasicInfo;
+import org.wso2.carbon.identity.role.v2.mgt.core.IdentityRoleManagementException;
+import org.wso2.carbon.identity.role.v2.mgt.core.IdpGroup;
+import org.wso2.carbon.identity.role.v2.mgt.core.Permission;
+import org.wso2.carbon.identity.role.v2.mgt.core.Role;
+import org.wso2.carbon.identity.role.v2.mgt.core.RoleBasicInfo;
+import org.wso2.carbon.identity.role.v2.mgt.core.UserBasicInfo;
+
+import java.util.List;
+
+/**
+ * Provides a set of methods to act as listeners before and after key operations related to role management.
+ * These listeners can be utilized to enforce additional checks, validations, or side-effects for
+ * role management functions.
+ */
+public interface RoleManagementListener {
+
+    /**
+     * Get the execution order identifier for this listener.
+     *
+     * @return The execution order identifier integer value.
+     */
+    int getExecutionOrderId();
+
+    /**
+     * Get the default order identifier for this listener.
+     *
+     * @return default order id
+     */
+    public int getDefaultOrderId();
+
+    /**
+     * Check whether the listener is enabled or not
+     *
+     * @return true if enabled
+     */
+    public boolean isEnable();
+
+    /**
+     * Invoked before a new role is added.
```

Comment:
Seems the comments in this file are not formatted correctly
Recheck whether wso2 code style is correctly configured https://github.com/wso2/product-is/wiki/Contributing-to-Identity-Server#how-to-configure-wso2-code-style-template
and format the file

Reply:
fixed